src/main/java/sonarquberepair/Main.java

```diff
@@ -1,66 +1,106 @@
+		opt.setDefault("2116");
+		opt.setHelp("Sonarqube rule number, Check https://rules.sonarsource.com/java");
+		jsap.registerParameter(opt);
+
+		opt = new FlaggedOption("projectKey");
+		opt.setLongFlag("projectKey");
+		opt.setStringParser(JSAP.STRING_PARSER);
+		opt.setDefault("fr.inria.gforge.spoon:spoon-core");
+		opt.setHelp("what is this projectKey ?");
+		jsap.registerParameter(opt);
+
+		opt = new FlaggedOption("originalFilesPath");
+		opt.setLongFlag("originalFilesPath");
+		opt.setStringParser(JSAP.STRING_PARSER);
+		opt.setDefault("./source/act/");
+		opt.setHelp("The input folder or file for SonarQube Repair to work on");
+		jsap.registerParameter(opt);
+
+		opt = new FlaggedOption("prettyPrintingStrategy");
+		opt.setLongFlag("prettyPrintingStrategy");
+		opt.setStringParser(JSAP.STRING_PARSER);
+		opt.setDefault(PrettyPrintingStrategy.NORMAL.name());
+		opt.setHelp("Mode for pretty printing . NORMAL: default pretty print, SNIPER: sniper mode on for more precise code transformation pretty print");
+		jsap.registerParameter(opt);
+
+		opt = new FlaggedOption("fileOutputStrategy");
+		opt.setLongFlag("fileOutputStrategy");
+		opt.setStringParser(JSAP.STRING_PARSER);
+		opt.setDefault(FileOutputStrategy.CHANGED_ONLY.name());
+		opt.setHelp("Mode for output. CHANGED_ONLY: default choice outputing only files modified by processors, ALL: everything including those unchanged files");
+		jsap.registerParameter(opt);
 
-	public static void repair(String pathToFile, String projectKey, int ruleKey, PrettyPrintingStrategy prettyPrintingStrategy) throws Exception {
-		Launcher launcher = new Launcher();
-		launcher.addInputResource(pathToFile);
-		launcher.getEnvironment().setAutoImports(true);
-		if (prettyPrintingStrategy == PrettyPrintingStrategy.SNIPER) {
-			launcher.getEnvironment().setPrettyPrinterCreator(() -> {
-						SniperJavaPrettyPrinter sniper = new SniperJavaPrettyPrinter(launcher.getEnvironment());
-						sniper.setIgnoreImplicit(false);
-						return sniper;
-					}
-			);
-			launcher.getEnvironment().setCommentEnabled(true);
-			launcher.getEnvironment().useTabulations(true);
-			launcher.getEnvironment().setTabulationSize(4);
-		}
-
-		Class<?> processor = Processors.getProcessor(ruleKey);
-		Constructor<?> cons;
-		Object object;
-		try {
-			cons = processor.getConstructor(String.class);
-			object = cons.newInstance(projectKey);
-		} catch (NoSuchMethodException e) {
-			cons = processor.getConstructor();
-			object = cons.newInstance();
-		}
-		launcher.addProcessor((Processor) object);
-		launcher.run();
+		opt = new FlaggedOption("workspace");
+		opt.setLongFlag("workspace");
+		opt.setStringParser(JSAP.STRING_PARSER);
+		opt.setDefault("./sonar-workspace");
+		opt.setHelp("Workspace of SonarQubeRepair");
+		jsap.registerParameter(opt);
+
+		opt = new FlaggedOption("gitRepoPath");
+		opt.setLongFlag("gitRepoPath");
+		opt.setStringParser(JSAP.STRING_PARSER);
+		opt.setHelp("Root Path of the Git repo directory");
```

Comment:
Which git repo directory?

Reply:
The originalFilesPath might not be the same as this. We might have multiple other subdir and only one of them is the Git repo dir which we need to specify where it is to generate a non-corrupt patch.